### PR TITLE
Fix sitemap

### LINKED ISSUE #22270: Generated sitemaps aren't valid according to Google Search Console
### Description

As of v1.18, Gitea now generates sitemaps. So I updated my Gitea instance and loaded its sitemap into Google Search Console, and it threw a bunch of errors.

(I can't reproduce with https://try.gitea.io because Google Search Console only allows me to load sitemaps from domains I control, and I definitely don't control https://try.gitea.io.)

### Gitea Version

v1.18

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

Screenshot 2022-12-29 at 18-48-09 Sitemaps

### Git Version

_No response_

### Operating System

Linux

### How are you running Gitea?

Docker Compose

### Database

SQLite

### PR DESCRIPTION
**Pull request by wolfogre**
Fix #22270.

Related to #18407.

The old code treated both sitemap and sitemap index as the format like:

```xml
...
<url>
  <loc>http://localhost:3000/explore/users/sitemap-1.xml</loc>
</url>
...
```

Actually, it's incorrect for sitemap index, it should be:

```xml
...
<sitemap>
  <loc>http://localhost:3000/explore/users/sitemap-1.xml</loc>
</sitemap>
...
```

See https://www.sitemaps.org/protocol.html

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: modules/sitemap/sitemap.go [hunk 1]
```diff
@@ -11,48 +11,62 @@ import (
...
-// Write writes the sitemap to a response
+// WriteTo writes the sitemap to a response
 func (s *Sitemap) WriteTo(w io.Writer) (int64, error) {
-	if len(s.URLs) > 50000 {
-		return 0, fmt.Errorf("The sitemap contains too many URLs: %d", len(s.URLs))
+	if l := len(s.URLs); l > urlsLimit {
+		return 0, fmt.Errorf("The sitemap contains too many URLs: %d", l)
+	}
+	if l := len(s.Sitemaps); l > urlsLimit {
+		return 0, fmt.Errorf("The sitemap contains too many URLs: %d", l)
```
Comment (delvh) [inline modules/sitemap/sitemap.go]:
```suggestion
		return 0, fmt.Errorf("The sitemap contains %d URLs, but only %d are allowed", l, urlsLimit)
	}
	if l := len(s.Sitemaps); l > urlsLimit {
		return 0, fmt.Errorf("The sitemap contains %d sub-sitemaps, but only %d are allowed", l, urlsLimit)
```

File: modules/sitemap/sitemap.go [hunk 2]
```diff
@@ -11,48 +11,62 @@ import (
...
 	}
```
Comment (delvh) [inline modules/sitemap/sitemap.go:70]: I also recommend changing the error message in `#79` to something similar.

**Review by lafriks — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Comment by jolheiser** [comment]
Please send backport. 🙂

**Pull request merged**